pkg/controller/controller.go

```diff
@@ -571,14 +585,21 @@ func resolveTemplate(template string, params map[string]string) (string, error)
 	return resolved, nil
 }
 
-func getCredentials(k8s kubernetes.Interface, ref *v1.SecretReference) (map[string]string, error) {
+func getCredentials(secrectsStore cache.Store, ref *v1.SecretReference) (map[string]string, error) {
 	if ref == nil {
 		return nil, nil
 	}
 
-	secret, err := k8s.CoreV1().Secrets(ref.Namespace).Get(ref.Name, metav1.GetOptions{})
+	secretObject, exists, err := secrectsStore.GetByKey(fmt.Sprintf("%s/%s", ref.Namespace, ref.Name))
 	if err != nil {
 		return nil, fmt.Errorf("error getting secret %s in namespace %s: %v", ref.Name, ref.Namespace, err)
+	} else if !exists {
+		return nil, fmt.Errorf("secret %s in namespace %s is not exists", ref.Name, ref.Namespace)
```

Comment:
Should read `secret %s in namespace %s does not exist`

Reply:
Ok.